OOInteraction/src/commands/AddNodeCommand.cpp

```diff
@@ -0,0 +1,85 @@
+ **
+ **    * Redistributions of source code must retain the above copyright notice, this list of conditions and the
+ **      following disclaimer.
+ **    * Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the
+ **      following disclaimer in the documentation and/or other materials provided with the distribution.
+ **    * Neither the name of the ETH Zurich nor the names of its contributors may be used to endorse or promote products
+ **      derived from this software without specific prior written permission.
+ **
+ **
+ ** THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+ ** INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ ** DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ ** SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ ** SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ ** WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ ** OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ **
+ **********************************************************************************************************************/
+
+#include "AddNodeCommand.h"
+#include "VisualizationBase/src/items/ViewItem.h"
+#include "OOModel/src/declarations/Project.h"
+
+namespace OOInteraction {
+
+AddNodeCommand::AddNodeCommand()
+	:MenuCommand("addNode", QStringList({"current", "0", "0"}))
+{
+}
+
+bool AddNodeCommand::canUseTarget(Visualization::Item *, Visualization::Item *target,
+		const std::unique_ptr<Visualization::Cursor>&)
+{
+	return target->hasNode() && dynamic_cast<OOModel::Method*>(target->node());
+}
+
+Interaction::CommandResult* AddNodeCommand::executeWithArguments(Visualization::Item *, Visualization::Item *target,
+		const QStringList& arguments, const std::unique_ptr<Visualization::Cursor>&)
+{
+	auto method = dynamic_cast<OOModel::Method*>(target->node());
+	auto name = arguments.at(0);
+	auto colOk = true;
+	auto rowOk = true;
+	auto column = arguments.at(1).toInt(&colOk);
+	auto row = arguments.at(2).toInt(&rowOk);
+	if (name == "current")
+		name = target->scene()->currentViewItem()->name();
+	auto view = target->scene()->getViewItem(name);
+	if (view && rowOk && colOk)
+	{
+		view->insertNode(method, column, row);
+		return new Interaction::CommandResult();
+	}
+	else if (!view)
+		return new Interaction::CommandResult(new Interaction::CommandError(
+											"The view with name " + name + " does not exist"));
+	else if (!colOk)
+		return new Interaction::CommandResult(new Interaction::CommandError(
+												"Could not parse " + arguments.at(1) + " to integer"));
```

Comment:
A better and shorter message here  and below is"
arguments.at(1) + " is not an integer"